hibernate/hibernate-orm/src/main/java/io/quarkus/qe/hibernate/hql/DevUIJsonRpcClient.java

```diff
@@ -0,0 +1,243 @@
+            return null;
+        } else {
+            if (loopCount > 10)
+                throw new RuntimeException("Too many recursions, message not returned for id [" + id + "]");
+
+            TimeUnit.SECONDS.sleep(3);
+            return objectResultFromJsonRPC(id, loopCount + 1);
+        }
+    }
+
+    private String createJsonRPCRequest(int id, String methodName, Map<String, Object> params) throws IOException {
+
+        ObjectNode request = mapper.createObjectNode();
+
+        request.put("jsonrpc", "2.0");
+        request.put("id", id);
+        request.put("method", this.namespace + "_" + methodName);
+        ObjectNode jsonParams = mapper.createObjectNode();
+        if (params != null && !params.isEmpty()) {
+            for (Map.Entry<String, Object> p : params.entrySet()) {
+                JsonNode convertValue = mapper.convertValue(p.getValue(), JsonNode.class);
+                jsonParams.putIfAbsent(p.getKey(), convertValue);
+            }
+        }
+        request.set("params", jsonParams);
+        return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(request);
+    }
+
+    private int sendRequest(String methodName, Map<String, Object> params) throws IOException {
+        Random random = new Random();
+        int id = random.nextInt(Integer.MAX_VALUE);
+        String request = createJsonRPCRequest(id, methodName, params);
+
+        Vertx vertx = Vertx.vertx();
+
+        HttpClientOptions options = new HttpClientOptions()
+                .setDefaultHost(this.uri.getHost())
+                .setDefaultPort(this.uri.getPort());
+
+        HttpClient client = vertx.createHttpClient(options);
+
+        WebSocketConnectOptions socketOptions = new WebSocketConnectOptions()
+                .setHost(this.uri.getHost())
+                .setPort(this.uri.getPort())
+                .setURI(this.uri.getPath());
+
+        client.webSocket(socketOptions, ar -> {
+            if (ar.succeeded()) {
+                WebSocket socket = ar.result();
+                Buffer accumulatedBuffer = Buffer.buffer();
+
+                socket.frameHandler((e) -> {
+                    Buffer b = accumulatedBuffer.appendBuffer(e.binaryData());
+                    if (e.isFinal()) {
+                        RESPONSES.put(id, new WebSocketResponse(b.toString()));
+                    }
+                });
+
+                socket.writeTextMessage(request);
+
+                socket.exceptionHandler((e) -> {
+                    RESPONSES.put(id, new WebSocketResponse(e));
+                    vertx.close();
+                });
+                socket.closeHandler(v -> {
+                    vertx.close();
+                });
+            } else {
+                RESPONSES.put(id, new WebSocketResponse(ar.cause()));
+                vertx.close();
+            }
+        });
+        return id;
+    }
+
+    private static final ConcurrentHashMap<Integer, WebSocketResponse> RESPONSES = new ConcurrentHashMap<>();
```

Comment:
I am not fan of this class, but we can keep it while it works, it makes no sense to waste more time on it. But this part doesn't make sense to me and I think it can confuse results. It is a static class, but the `HQLConsoleClient` creates `new DevUIJsonRpcClient("quarkus-hibernate-orm", testUrl)`, so why is it not a member property instead?

The issue is that I know that at least in some cases, JUnit initialize test classes together (I don't know if it is every case or whatever) because I was fixing in a framework issue that one test class had stored state in static variables from other test class. can you try to keep the `RESPONSES` without `static` and discover why is it needed?